src/Angor/Avalonia/AngorApp/Sections/Browse/Details/Invest/Amount/AmountViewModel.cs

```diff
@@ -1,29 +1,34 @@
-using System.Reactive.Linq;
-using Angor.UI.Model;
+using Angor.Wallet.Application;
+using Angor.Wallet.Domain;
 using ReactiveUI.SourceGenerators;
 using ReactiveUI.Validation.Extensions;
 using ReactiveUI.Validation.Helpers;
+using Zafiro.CSharpFunctionalExtensions;
 
 namespace AngorApp.Sections.Browse.Details.Invest.Amount;
 
 public partial class AmountViewModel : ReactiveValidationObject, IAmountViewModel
 {
     [Reactive] private long? amount;
 
-    public AmountViewModel(IWallet wallet, IProject project)
+    public AmountViewModel(WalletId walletId, IWalletAppService walletAppService, IProject project)
     {
-        walletBalanceHelper = wallet.Balance.ToProperty(this, model => model.WalletBalance);
+        LoadBalance = ReactiveCommand.CreateFromObservable(() => Observable.FromAsync(() => walletAppService.GetBalance(walletId)).Successes().Select(x => x.Value));
```

Comment:
Hi J, check out this new service method I made for wallet refresh, it might be best to create something similar on the shared project?

https://github.com/block-core/angor/blob/main/src/Angor/Client/Services/WalletUIService.cs#L41

Reply:
Nice! I'll take a look and implement it in a new PR. Thanks for the tip!